crates/forge-runner/src/running.rs

```diff
@@ -265,15 +293,50 @@ pub fn run_test_case(
         None
     };
 
+    // copied from casm_run
```

Comment:
Why do we need to copy this from casm_run?
I thought it would be possible just to put CairoRunner inside some function and then call finalize on it and vm

Reply:
I've tried to minimize changes to the casm_run interface, so we have to "copy" these lines:

```rust
    let mut vm = VirtualMachine::new(true);
    let data: Vec<MaybeRelocatable> = instructions
        .flat_map(|inst| inst.assemble().encode())
        .map(Felt252::from)
        .map(MaybeRelocatable::from)
        .collect();
    let data_len = data.len();
```

that is preparation of data to be passed to the runner